src/main/java/Duke.java

```diff
@@ -1,10 +1,136 @@
+import java.io.File;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.ArrayList;
+
 public class Duke {
+    private SaveToFile saveToFile;
+    private Ui ui;
+    private TaskList tasks;
+
+    /**
+     * Constructor for duke.
+     *
+     * @param filePath File path for duke's task list.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        saveToFile = new SaveToFile();
+        ArrayList<Task> arr = new ArrayList<>();
+        tasks = new TaskList(saveToFile.loadList(filePath));
+    }
+
+    /**
+     * Run duke
+     */
+    public void run() {
+        ui.showWelcome();
+        Scanner sc = new Scanner(System.in);
+
+        while (sc.hasNext()) {
+            String userInput = sc.nextLine();
+            String[] arrString = userInput.split(" ", 2);
+            try {
+                if (arrString[0].equalsIgnoreCase("bye")) {
+                    ui.goodbyeMessage();
+                    break;
+                } else if (arrString[0].equalsIgnoreCase("list")) {
```

Comment:
I think this part can be better refactored, it looks like arrow-head style and feels really long and wordy to read. Like for example you can leave printing things purely to the Ui and just pass it the string that needs to be printed.